src/ralph/virtual/admin.py

```diff
@@ -52,11 +54,30 @@ class VirtualServerTypeForm(RalphAdmin):
 
 
 class VirtualServerForm(RalphAdminForm):
+    HYPERVISOR_TYPE_ERR_MSG = 'Hypervisor must be one of \
+                                   DataCenterAsset or VirtualServer.'
+    HYPERVISOR_IS_SELF_ERR_MSG = 'Hypervisor and host must be different.'
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if 'parent' in self.fields:
             self.fields['parent'].required = True
 
+    def is_valid(self):
```

Comment:
isn't it possible to use (create) `validate_parent` method?

Reply:
Done, now it's a lot cleaner.